### PR TITLE
[SYCL] Adds regression test for reduction resource leak

### PR DESCRIPTION
**Pull request by steffenlarsen**
Select variants of reductions currently leak additional resources, such as auxiliary buffers. This commit adds a regression test to ensure this leak does not resurface.

Fix: https://github.com/intel/llvm/pull/5653

### CODE REVIEW

File: SYCL/Regression/reduction_resource_leak.cpp [hunk 1]
```diff
@@ -0,0 +1,30 @@
+// REQUIRES: level_zero
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: env ZE_DEBUG=4 %GPU_RUN_PLACEHOLDER %t.out 2>&1 | FileCheck --implicit-check-not "LEAK" %s
+
+// Tests that additional resources required by discard_write reductions od not
```
Comment (romanovvlad) [inline SYCL/Regression/reduction_resource_leak.cpp]:
```suggestion
// Tests that additional resources required by discard_write reductions do not
```

**Review by romanovvlad — DISMISSED** [review]
LGTM

**Comment by steffenlarsen** [comment]
https://github.com/intel/llvm/pull/5653 has been merged and verification with it passed. @vladimirlaz & @romanovvlad - Could you please have another look at this?

**Review by vladimirlaz — DISMISSED** [review]

**Comment by romanovvlad** [comment]
> [intel/llvm#5653](https://github.com/intel/llvm/pull/5653) has been merged and verification with it passed. @vladimirlaz & @romanovvlad - Could you please have another look at this?

Look OK, but don't we want to test USM case?

**Comment by steffenlarsen** [comment]
@romanovvlad - I have added a USM test case.

**Review by romanovvlad — APPROVED** [review]

**Pull request merged**